app/authenticate/views.py

```diff
@@ -0,0 +1,23 @@
+from django.shortcuts import render, redirect
+from django.contrib.auth.models import Group
+from django.contrib.auth import login, logout, authenticate
+# from django.contrib.auth.forms import UserCreationForm
+from django.urls import reverse_lazy
+from django.views import generic
+from django.contrib.auth.decorators import login_required
+from .forms import SignUpForm
+#
+def user_register(request):
+    if request.method == 'POST':
+        form = SignUpForm(request.POST)
+        if form.is_valid():
+            form.save()
+            username = form.cleaned_data.get('username')
+            request.session['username'] = username
+            raw_password = form.cleaned_data.get('password1')
+            user = authenticate(username=username, password=raw_password)
+            login(request, user)
+            return render(request, 'homepage/homepage.html')
```

Comment:
Rather than using `render`, should we use `redirect` instead? For example:

```
from homepage.views import homepage

def user_register(request):
    ...
    return redirect(homepage)
```

With render, the URL in the address bar remains at `/register/`, whereas with redirect it triggers a new request for the homepage at the site root.

In the future, we may want to include a 'next' attribute which remember which page the user came from then redirects them back to there. Something to think about in the future.